pkg/authn/oidc.go

```diff
@@ -36,20 +37,30 @@ var (
 )
 
 // NewOIDCAuthenticator returns OIDC authenticator
-func NewOIDCAuthenticator(config *OIDCConfig) (*OIDCAuthenticator, error) {
+func NewOIDCAuthenticator(ctx context.Context, config *OIDCConfig) (*OIDCAuthenticator, error) {
 	dyCA, err := dynamiccertificates.NewDynamicCAContentFromFile("oidc-ca", config.CAFile)
 	if err != nil {
 		return nil, err
 	}
 
-	tokenAuthenticator, err := oidc.New(oidc.Options{
-		IssuerURL:            config.IssuerURL,
-		ClientID:             config.ClientID,
+	tokenAuthenticator, err := oidc.New(ctx, oidc.Options{
+		JWTAuthenticator: apiserver.JWTAuthenticator{
```

Comment:
do we get the dynamic CA for free when we move it here instead of using it as below?

Reply:
Which of the properties enables a dynamic CA from within JWTAuthenticator?

Reply:
The dynamic CA content provider is being used to create the client.

https://github.com/kubernetes/kubernetes/blob/9d5db28f5f5d5a35012dd6162af374420e6e1f4c/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go#L270-L291